client/src/backoffice/EventsPage.tsx

```diff
@@ -117,49 +154,130 @@ const EventForm: React.FC<{ create?: boolean; event?: PoapEvent }> = ({ create,
 
-            <SubmitButton text="Save" isSubmitting={isSubmitting} canSubmit={dirty && isValid} />
+            <ImageContainer
+              text="Upload Image:"
+              handleFileChange={handleFileChange}
+              setFieldValue={setFieldValue}
+              errors={errors}
+            />
+            {event && typeof event.image === 'string' && (
+              <div className={'image-edit-container'}>
+                <img className={'image-edit'} src={event.image} />
+              </div>
+            )}
+            <SubmitButton text="Save" isSubmitting={isSubmitting} canSubmit={true} />
           </Form>
         )}
       </Formik>
-      {!create && <Link to={`/claim/${event!.fancy_id}`}>Go to Claim Page</Link>}
     </div>
   );
 };
 
+type DatePickerContainerProps = {
+  text: string;
+  dayToSetup: 'start_date' | 'end_date';
+  handleDayClick: Function;
+  setFieldValue: Function;
+  disabledDays: RangeModifier | undefined;
+};
+export interface RangeModifier {
+  from: Date;
+  to: Date;
+}
+
+const DayPickerContainer = ({
+  text,
+  dayToSetup,
+  handleDayClick,
+  setFieldValue,
+  disabledDays,
+}: DatePickerContainerProps) => (
+  <div className="date-picker-container">
+    <span>{text}</span>
+    <DayPickerInput
+      dayPickerProps={{ disabledDays: disabledDays }}
+      onDayChange={(day: Date) => handleDayClick(day, dayToSetup, setFieldValue)}
+    />
+  </div>
+);
+
+type ImageContainerProps = {
+  text: string;
+  handleFileChange: Function;
+  setFieldValue: Function;
+  errors: any;
+};
+
+const ImageContainer = ({ text, handleFileChange, setFieldValue, errors }: ImageContainerProps) => (
+  <div className="date-picker-container">
+    <span>{text}</span>
+    <input
+      type="file"
+      className={classNames(Boolean(errors.image) && 'error')}
+      onChange={(e: ChangeEvent<HTMLInputElement>) => handleFileChange(e, setFieldValue)}
+    />
+    <ErrorMessage name="image" component="p" className="bk-error" />
+  </div>
+);
+
 type EventFieldProps = {
   title: string;
   name: string;
   disabled?: boolean;
+  type?: string;
 };
-const EventField: React.FC<EventFieldProps> = ({ title, name, disabled }) => {
+const EventField: React.FC<EventFieldProps> = ({ title, name, disabled, type }) => {
   return (
     <Field
       name={name}
       render={({ field, form }: FieldProps) => (
         <div className="bk-form-row">
           <label>{title}:</label>
-          <input
-            type="text"
-            {...field}
-            disabled={disabled}
-            className={classNames(!!form.errors[name] && 'error')}
-          />
+          {type === 'textarea' && (
+            <textarea
+              {...field}
+              wrap="soft"
+              disabled={disabled}
+              className={classNames(!!form.errors[name] && 'error')}
+            />
+          )}
+          {type !== 'textarea' && (
+            <input
+              type={type || 'text'}
+              {...field}
```

Comment:
Should the spread be in the first line in case it has a `type`? What I mean is that if the `field` has a `type` property it will have more priority / will step our `type={type || 'text'}`

Reply:
right! i swapped their positions